backend/modules/invoices.js

```diff
@@ -63,8 +84,49 @@ router.get('/file/:id', async (req, res) => {
     res.send(file);
   } catch (e) {
     console.error(e);
-    res.status(500).json({success: false, message: 'Internal Server Error'});
+    res.status(500).json({ success: false, message: 'Internal Server Error !' });
   }
 });
 
+// Endpoint to add a new invoice
+app.post('/invoice/new', checkRole, (req, res) => {
+  const { amount, eventId } = req.query;
+
+  // amount and eventId are necessary
+  if (!amount || !eventId) {
+    return res.status(400).json({ error: 'amount and eventId are required in query parameters' });
+  }
+
+  // Create a new invoice 
+  const newInvoice = {
+    id: invoices.length + 1,
+    amount: parseFloat(amount),
+    eventId: eventId,
+    status: 'pending'
+  };
+
+  // Push the new invoice to the database 
+  invoices.push(newInvoice);
```

Comment:
What is `invoices` variable here? To push to database write a database query using `pool.query`. Ask Ishita for help here.